tap-gui/tap-gui-rbac/set-up-tap-gui-permission-framework.hbs.md

```diff
@@ -0,0 +1,125 @@
+
+## <a id='permission-overview'></a> Overview of the permission framework
+
+Permission framework allows Tanzu Application Platform operators to enforce policies that would limit visibility of certain parts of the Tanzu Application Platform GUI. 
+
+The current release features the `alpha` version of this functionality and only applies to the Software Catalog entities. Only one policy is available and it is embedded in the GUI.
+
+The policy is called `owner-of`. As it is enabled, the operator, can define admin groups and users that would have unrestricted visibility of the Software Catalog entities. If a user is not listed as an admin or part of the admin group, they would be able to see onle the Software Catalog entities that are owned by teams they belong to.
+
+**Note** Permission framework is disabled by default.
+
+## <a id='enable-permission'></a> Enable the permission framework
+
+To enable the permission framework and limit Software Catalog entities' visibility to owners, allow-listed users/groups, and catalog admins, please add the following parameters to the `tap_gui.app_config` stanza:
+
+```yaml
+permission:
+  enabled: true
+  adminRefs:
+    - user:<namespace>/<name>
+    - group:<namespace>/<name>
+```
+Where `adminRefs` section lists all users and groups that should be granted access. Each user and group should be defined by the
+* `<namespace>`, which usually equals to `default` unless defined otherwise in the definition file, and
+* `<name>` is the name of the group or user.
+
+You can view your User Entity in the `Settings`/`General` section in your Tanzu Application Platform GUI, or in the yaml definition file of the User or Group.
+
+For example:
+```yaml
+- user:default/admin
+- group:test-namespace/operators
+```
+
+Once you have updated your configuration file, reinstall your Tanzu Application Platform GUI package by following the steps in
+   [Upgrade Tanzu Application Platform](../../upgrading.hbs.md).
+
+After the updated values configuration file is applied in Tanzu Application Platform GUI, the permission framework shall become enabled.
+
+**Note** Once the permission framework is enabled, all Software Catalog entities without proper annotations shall become invisible to all users except admins.
+
+## <a id='enable-visibility'></a> Enable catalog entity visibility
+
+With the permission framework enabled, it is required to annotate entities in your Software Catalog to give visibility to the required Users or Groups. To do that, you will need to add annotations to the `metadata.annotations` stanza of the entity's definition file.
+
+```yaml
+metadata:
+  annotations:
+    backstage.tanzu.vmware.com/<group|user>.<namespace>.<name>: 'COMMA-SEPARATED-PERMISSIONS'
```

Comment:
@LittleBaiBai can the user add several annotations? If yes, will each one be respected by the policy?

Reply:
Yes, as long as it's one annotation per user/group (example below), the permission framework will work.
```
metadata:
  annotations:
    backstage.tanzu.vmware.com/user.default.user-a: 'catalog.entities.read'
    backstage.tanzu.vmware.com/user.default.user-b: 'catalog.entities.read,catalog.entities.update'
```